core/src/main/kotlin/org/eln2/sim/thermal/main.kt

```diff
@@ -0,0 +1,146 @@
+import kotlinx.coroutines.*
+import kotlinx.coroutines.sync.*
+import kotlin.math.*
+
+class ThermalInterface{
+
+    var mutex = Mutex()
+    var R = 1f //heat resistance at the interface
+    var heatTranfer = 0f //heat transfer during a simulation tick through this interface
+
+    var lastTickComputed = 0 //number of the last tick where the heat transfer was computed. Tick number switch between 0 and 1. This is here to prevent computing twice the heat trasnfered when not needed
+
+
+    fun computeHeatTransfer(dt : Float, tickNumber : Int){
+        var T0 = e0.getT(tickNumber)
+        var T1 = e1.getT(tickNumber)
+        if(tickNumber != lastTickComputed) {
+            heatTranfer = (T1 - T0) / R * dt
+            lastTickComputed = tickNumber
+        }
+    }
+
+    lateinit var e0 : ThermalElement
+    lateinit var e1 : ThermalElement
+}
+
+class SignedThermalInterface(thInterface : ThermalInterface, sgn : Float){
```

Comment:
You can integrate these fields right into the constructor parameters if you don't need to do anything fancy with them:

```kotlin
class SignedThermalInterface(val thermalInterface: ThermalInterface, val sign: Float) {
    fun computeHeatTransfer(dt: Float, tickNumber: Int): Float {
        // ...
    }
}
```

Reply:
I had no idea you could do that! Is that the common practice?

Reply:
I can't speak to how common it is in the wild, just acquainting myself with Kotlin for this project, but [it](https://github.com/eln2/eln2/blob/8a0b670a7c6bc6018e9db09a1a94aad6c32566c8/core/src/main/kotlin/org/eln2/sim/electrical/mna/Node.kt#L12) [certainly](https://github.com/eln2/eln2/blob/8a0b670a7c6bc6018e9db09a1a94aad6c32566c8/core/src/main/kotlin/org/eln2/data/MultiMap.kt#L6) [seems](https://github.com/eln2/eln2/blob/8a0b670a7c6bc6018e9db09a1a94aad6c32566c8/core/src/main/kotlin/org/eln2/space/Location.kt#L8) [useful](https://github.com/eln2/eln2/blob/8a0b670a7c6bc6018e9db09a1a94aad6c32566c8/core/src/main/kotlin/org/eln2/space/Location.kt#L41) [when](https://github.com/eln2/eln2/blob/8a0b670a7c6bc6018e9db09a1a94aad6c32566c8/core/src/main/kotlin/org/eln2/space/Location.kt#L309) [making](https://github.com/eln2/eln2/blob/8a0b670a7c6bc6018e9db09a1a94aad6c32566c8/core/src/main/kotlin/org/eln2/space/Graph.kt#L9) [containers](https://github.com/eln2/eln2/blob/8a0b670a7c6bc6018e9db09a1a94aad6c32566c8/core/src/main/kotlin/org/eln2/space/Graph.kt#L29) [or](https://github.com/eln2/eln2/blob/8a0b670a7c6bc6018e9db09a1a94aad6c32566c8/core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt#L38) [wrappers](https://github.com/eln2/eln2/blob/8a0b670a7c6bc6018e9db09a1a94aad6c32566c8/core/src/main/kotlin/org/eln2/parsers/falstad/Falstad.kt#L50). Definitely a good thing to know about :)